addons/website/views/website_templates.xml

```diff
@@ -200,8 +200,32 @@
             <script>
                 window.dataLayer = window.dataLayer || [];
                 function gtag(){dataLayer.push(arguments);}
+                gtag('consent', 'default', {
+                    'ad_storage': 'denied',
+                    'ad_user_data': 'denied',
+                    'ad_personalization': 'denied',
+                    'analytics_storage': 'denied',
+                });
                 gtag('js', new Date());
                 gtag('config', '<t t-esc="website.google_analytics_key"/>');
+                function allConsentGranted() {
+                    gtag('consent', 'update', {
+                        'ad_storage': 'granted',
+                        'ad_user_data': 'granted',
+                        'ad_personalization': 'granted',
+                        'analytics_storage': 'granted',
+                    });
+                }
+                <t t-if="not website.cookies_bar or request.env['ir.http']._is_allowed_cookie('optional')">
+                    allConsentGranted();
+                </t>
+                <t t-else="">
+                    document.addEventListener(
+                        "optionalCookiesAccepted",
+                        allConsentGranted,
```

Comment:
@robinlej You only send a `"consent > update"` if optional cookies are accepted (or immediately when they are accepted). Which means that when a user clicks on `"only allow essential cookies"`, only the `"consent > default"` is sent without any update. Should we send a `"denied"` update in this case ?